scripts/compile.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const {EOL} = require('os');
+const path = require('path');
+const ts = require('typescript');
+
+/**
+ * @param {string} tsConfigPath
+ * @param {string} folder
+ */
+function compile(tsConfigPath, folder) {
+  const tsConfig = ts.readConfigFile(tsConfigPath, path => ts.sys.readFile(path));
+
+  if (tsConfig.error) {
+    console.error(ts.formatDiagnostic(tsConfig.error, getFormatDiagnosticsHost(folder)));
+    return 1;
+  }
+
+  const parsedConfig = ts.parseJsonConfigFileContent(tsConfig.config, ts.sys, folder);
+
+  parsedConfig.options.rootDir = path.resolve(folder, 'sources');
+  parsedConfig.options.outDir = path.resolve(folder, 'lib');
+
+  const program = ts.createProgram({
+    options: parsedConfig.options,
+    rootNames: parsedConfig.fileNames,
+    configFileParsingDiagnostics: parsedConfig.errors,
+  });
+
+  const diagnostics = program.emit();
+
+  return reportErrors(diagnostics.diagnostics, folder);
+}
+exports.compile = compile;
+
+/**
+ * @param {readonly import('typescript').Diagnostic[]} allDiagnostics
+ * @param {string} folder
+ */
+function reportErrors(allDiagnostics, folder) {
+  const errorsAndWarnings = allDiagnostics.filter(function(d) {
+    return d.category !== ts.DiagnosticCategory.Message;
+  });
+
+  if (errorsAndWarnings.length === 0)
+    return 0;
+
+  const formatDiagnosticsHost = getFormatDiagnosticsHost(folder);
+
+  for (const errorAndWarning of errorsAndWarnings)
+    console.error(ts.formatDiagnostic(errorAndWarning, formatDiagnosticsHost));
+
+  return 1;
+}
+
+/**
+ * @param {string} folder
+ * @returns {import('typescript').FormatDiagnosticsHost}
+ */
+function getFormatDiagnosticsHost(folder) {
+  return {
+    getCurrentDirectory: () => folder,
```

Comment:
Passing the folder being built leads to errors that are logged relative to the package folder, like in the OP

```
sources/commands/workspaces/list.ts(15,10): error TS2742: ...
```

An alternative would be to use the repository root, to log the path of the file in the repo instead:

```
packages/plugin-essentials/sources/commands/workspaces/list.ts(15,10): error TS2742: ...
```

Any opinions?

Reply:
Slight preference on the second one to avoid ambiguities with `index.ts` etc